spec/unit/file_system_spec.rb

```diff
@@ -134,6 +134,36 @@ def increment_counter_in_multiple_processes(file, num_procs, options)
     end
   end
 
+  context "read should allow an encoding to be specified" do
+    # First line of Rune version of Rune poem at http://www.columbia.edu/~fdc/utf8/
+    # characters chosen since they will not parse on Windows with codepage 437 or 1252
+    let (:rune_utf8) { "\u16A0\u16C7\u16BB" } # 'ᚠᛇᚻ'
+
+    before :each do
+      # verify Ruby has treated string as UTF-8
+      expect(rune_utf8.encoding).to eq(Encoding::UTF_8)
```

Comment:
I was thinking that a ruby magic comment could cause ruby to use a different encoding for that string, but it doesn't appear to be the case:

```ruby
# Encoding: cp1252
puts "ᚠᛇᚻ".encoding
puts "\u16A0\u16C7\u16BB".encoding
```

Results in:

```
C:\Program Files\Puppet Labs\Puppet> chcp 1252
Active code page: 1252
C:\Program Files\Puppet Labs\Puppet> ruby C:\string.rb
Windows-1252
UTF-8
```

The thing I don't know is whether it guaranteed that the "\u..." string will internally be represented as UTF-8? For example, if I specify the default internal encoding, the "\u..." string is still `UTF-8`:

```
C:\Program Files\Puppet Labs\Puppet> ruby -E cp1252 C:\string.rb
Windows-1252
UTF-8
```

So long story short, I think what you have is fine.